Dshop/apps/products_catalogue/views.py

```diff
@@ -10,6 +10,15 @@ class ProductListView(ListView):
 
     def get_queryset(self):
         return Product.objects.filter(is_active=True)
+    
+class ProductDetailView(DetailView):
+    
+    model = Product
+    template_name = 'products_catalogue/product_detail.html'
 
+    def get_object(self):
+        slug = self.kwargs.get('slug')
+        product = Product.objects.get(slug=slug)
```

Comment:
Please update as per id in url, above. Slug will not be needed in this query. Actually, I think when you update url with the ID as above, the whole get_object method can be removed.

Reply:
Second issue here is that it does not use "is_active" flag, so it will show products that should not be shown. 
I would suggest removing this method and changing line 16 from model to Queryset as described in https://akademiait.com.pl/class-based-views-w-django-wygodny-sposob-tworzenia-widokow-czesc-1/ Detaiview section.